packages/components/pie-list/src/index.ts

```diff
@@ -12,11 +18,53 @@ const componentSelector = 'pie-list';
 
 /**
  * @tagname pie-list
+ *
+ * PIE List component for displaying collections of related items.
+ *
+ * @slot - List items to render inside the component. Supports `<pie-list-item>`
+ * elements or custom markup that behaves like a native list item.
  */
 @safeCustomElement('pie-list')
 export class PieList extends RtlMixin(PieElement) implements ListProps {
+    @property({ type: String, reflect: true })
+    public variant: ListVariant = defaultProps.variant;
+
+    @property({
+        type: Boolean,
+        reflect: true,
+        attribute: 'has-dividers',
+    })
+    public hasDividers = defaultProps.hasDividers;
+
+    @property({
+        type: String,
+        reflect: true,
+        attribute: 'list-type',
+    })
+    public listType: ListType = defaultProps.listType;
+
+    connectedCallback () {
+        super.connectedCallback();
+
+        if (!this.hasAttribute('data-test-id')) {
+            this.setAttribute('data-test-id', 'pie-list');
+        }
```

Comment:
could you please share more why we need to do this step? i think it's common for users to set a custom data-test-id on the host element and that will always override it